nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/AddPermission.java

```diff
@@ -110,8 +114,8 @@ protected void addPermission(DocumentModel doc) {
         }
 
         String username = user;
-        if (username == null) {
-            username = NuxeoPrincipal.computeTransientUsername(email);
+        if (!isUserFound(username)) {
```

Comment:
What if `user` is `null` but the `email` is not? This operation allows creating transient permissions for a user given only an `email`.
Seems we are missing a test here... (my fault, will add one after).

You should do your check only if  `user` is != `null`, otherwise keep the existing code creating a TransientUsername.